src/main/java/org/mitre/synthea/export/FhirR4.java

```diff
@@ -303,6 +341,43 @@ private static Table<String, String, String> loadMapping(String filename) {
     return mappingTable;
   }
 
+  private static Map<String, Table<String, String, String>>
+      loadMappingWithVersions(String filename, String... supportedVersions) {
+    Map<String, Table<String,String,String>> versions = new HashMap<>();
+
+    for (String version : supportedVersions) {
+      versions.put(version, HashBasedTable.create());
+    }
+
+    List<LinkedHashMap<String, String>> csvData;
+    try {
+      csvData = SimpleCSV.parse(Utilities.readResource(filename));
+    } catch (IOException e) {
+      e.printStackTrace();
+      return null;
+    }
+
+    for (LinkedHashMap<String, String> line : csvData) {
+      String system = line.get("SYSTEM");
+      String code = line.get("CODE");
+      String url = line.get("URL");
+      String version = line.get("VERSION");
+
+      if (StringUtils.isBlank(version)) {
+        // blank means applies to ALL versions
+        versions.values().forEach(table -> table.put(system, code, url));
+      } else {
+        Table<String, String, String> mappingTable = versions.get(version);
+        if (mappingTable == null) {
+          throw new IllegalArgumentException("Unknown version " + version);
+        }
```

Comment:
Perhaps a more informative exception message would be helpful.

Maybe, "Unknown FHIR US Core version number: " or something like that.

Reply:
Updated to include more detail including the filename and what versions are expected